dataeng/jobs/analytics/RetirementJobs.groovy

```diff
@@ -0,0 +1,725 @@
+            // but customer support can read and discover.
+            authorization {
+                blocksInheritance(true)
+                // Only core teams can run the retirement collector directly.
+                List membersWithFullControl = ['edx*edx-data-engineering', 'edx*testeng', 'edx*devops']
+                membersWithFullControl.each { emp ->
+                    permissionAll(emp)
+                }
+                // Other engineering teams can view.
+                List extraMembersCanView = ['edx*educator-all', 'edx*learner', 'bbaker6225']
+                extraMembersCanView.each { emp ->
+                    permission('hudson.model.Item.Read', emp)
+                    permission('hudson.model.Item.Discover', emp)
+                }
+                // TODO PLAT-2036: uncomment the following two lines when we add the
+                // appropriate github group.
+                //permission('hudson.model.Item.Read', 'edx/customer-support')
+                //permission('hudson.model.Item.Discover', 'edx/customer-support')
+            }
+
+            // Allow this job to have simultaneous builds at the same time in general,
+            // but use the throttle-concurrents plugin to limit only one instance of
+            // this job to run concurrently per environment.  This just helps keep
+            // things simple.
+            concurrentBuild(true)
+            throttleConcurrentBuilds {
+                // A maxTotal of 0 implies unlimited simultaneous jobs, but below we
+                // restrict one build per environment.
+                maxTotal(0)
+            }
+            configure { project ->
+                project / 'properties' / 'hudson.plugins.throttleconcurrents.ThrottleJobProperty' <<
+                    'paramsToUseForLimit'('ENVIRONMENT')
+                project / 'properties' / 'hudson.plugins.throttleconcurrents.ThrottleJobProperty' <<
+                    'limitOneJobWithMatchingParams'('true')
+            }
+
+            // keep jobs around for 30 days
+            // allVars contains the value for DAYS_TO_KEEP_BUILD which will be used inside AnalyticsConstants.common_log_rotator
+            logRotator common_log_rotator(allVars)
+
+            wrappers {
+                timeout {
+                    absolute(60*24)  // 24 hours
+                    failBuild()
+                }
+                buildUserVars() /* gives us access to BUILD_USER_ID, among other things */
+                // DENG-1063/PSRE-1204: Temporarily avoid using the build-name-setter
+                // plugin while it's currently broken due to plugin incompatibilities.
+                //buildName('#${BUILD_NUMBER}, ${ENV,var="ENVIRONMENT"}')
+                timestamps()
+                colorizeOutput('xterm')
+                credentialsBinding {
+                    file('USER_RETIREMENT_SECURE_DEFAULT', 'user-retirement-secure-default.yml')
```

Comment:
I think we can do away with this credential. It hasn't been used in years, and I think it was only used for development.  I would just make the ENVIRONMENT parameter a required parameter with no default.

Reply:
Sorry I couldn't understand the second part of comment (`I would just make the ENVIRONMENT parameter a required parameter with no default`)
I removed the line `file('USER_RETIREMENT_SECURE_DEFAULT', 'user-retirement-secure-default.yml')` so it gave me error on cp command as its being used in shell files of the job like here: [https://github.com/edx/jenkins-job-dsl/blob/master/platform/resources/user-retirement-collector.sh#L26](https://github.com/edx/jenkins-job-dsl/blob/master/platform/resources/user-retirement-collector.sh#L26)

Reply:
or can I just simply remove this [cp command](https://github.com/edx/jenkins-job-dsl/blob/master/platform/resources/user-retirement-collector.sh#L26) if this is not used anywhere ?

Reply:
Yes, I would say remove the cp command, remove the empty file credential, and remove the default value for the ENVIRONMENT parameter because all of the upstream trigger jobs will provide a value for ENVIRONMENT.

I just remembered why we added all of that logic---it was for the open source community!  However, the open source instances that have contacted us about user retirement haven't been using these jenkins jobs---they have written their own automation to call the tubular scripts directly.  I think we can narrow down our supported methods of supplying configuration to just what we currently do (fetch from analytics-secure).